app/src/main/java/de/xikolo/storages/RecentCoursesStorage.kt

```diff
@@ -0,0 +1,55 @@
+package de.xikolo.storages
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import de.xikolo.App
+import de.xikolo.storages.base.BaseStorage
+import de.xikolo.storages.base.SharedPreferenceLiveData
+
+class RecentCoursesStorage : BaseStorage(PREF_RECENT_COURSES, Context.MODE_PRIVATE) {
+
+    val recentCourses: LinkedHashSet<Pair<String, String>>?
+        get() {
+            val json = getString(PREF_RECENT_COURSES)
+            val type = object : TypeToken<LinkedHashSet<Pair<String, String>>>() {}.type
+            return Gson().fromJson(json, type)
+        }
+
+    private val shortcutMax = 3
+    val coursesLive: SharedPreferenceLiveData<String> by lazy {
+        // what name does the preference have? What is the sharedPref Parameter supposed to do?
```

Comment:
I'm supposed to use a name for a preference as a parameter. But is this free for me to decide how this preference is to be named? Other than that I'm not sure if a chose the correct sharedPreferences as a parameter.